### PR TITLE
[ISSUE #3721]Do some code optimization[JaegerTraceService]

### LINKED ISSUE #3721: [Enhancement] Do some code optimization[JaegerTraceService]
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/eventmesh/issues?q=is%3Aissue) and found no similar issues.

### Enhancement Request

image
located at:
eventmesh-trace-plugin/eventmesh-trace-jaeger/src/main/java/org/apache/eventmesh/trace/jaeger/JaegerTraceService.java line 126,127,132,133
analysis and explanation:
a. Not annotated parameter overrides @ParametersAreNonnullByDefault parameter
b. A "NullPointerException" could be thrown; "carrier" is nullable here.

### Describe the solution you'd like

add `@Nullable` and use Objects.repuireNonNull().
image

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### PR DESCRIPTION
**Pull request by pandaapo**
Fixes #3721.

### Motivation

Aoid `Not annotated parameter overrides @ParametersAreNonnullByDefault`
Correct a wrong variable in anonymous inner class in `ZipkinTraceService#inject()`.

### Documentation

- Does this pull request introduce a new feature? no

### CODE REVIEW

File: eventmesh-trace-plugin/eventmesh-trace-zipkin/src/main/java/org/apache/eventmesh/trace/zipkin/ZipkinTraceService.java [hunk 1]
```diff
@@ -144,10 +143,9 @@ public String get(Map<String, Object> carrier, String key) {
 
     @Override
     public void inject(Context context, Map<String, Object> map) {
-        textMapPropagator.inject(context, map, new TextMapSetter<Map<String, Object>>() {
-            @Override
-            public void set(@Nullable Map<String, Object> carrier, String key, String value) {
-                map.put(key, value);
+        textMapPropagator.inject(context, map, (cr, key, value) -> {
```
Comment (Alonexc) [inline eventmesh-trace-plugin/eventmesh-trace-zipkin/src/main/java/org/apache/eventmesh/trace/zipkin/ZipkinTraceService.java:146]: An issue has been raised here, and a pr has been submitted.issue #3480

Comment (pandaapo) [reply eventmesh-trace-plugin/eventmesh-trace-zipkin/src/main/java/org/apache/eventmesh/trace/zipkin/ZipkinTraceService.java:146]: Thanks for your review. That pr only replace with lambda, but  it seems that the variable `map` in lambda also should be modified.
https://github.com/apache/eventmesh/blob/124ec9f17897070afff85524596b594cec6d3bc1/eventmesh-trace-plugin/eventmesh-trace-zipkin/src/main/java/org/apache/eventmesh/trace/zipkin/ZipkinTraceService.java#L147

Comment (Alonexc) [reply eventmesh-trace-plugin/eventmesh-trace-zipkin/src/main/java/org/apache/eventmesh/trace/zipkin/ZipkinTraceService.java:146]: You can comment on the changes under that pr. You are very welcome to discuss it together.

Comment (pandaapo) [reply eventmesh-trace-plugin/eventmesh-trace-zipkin/src/main/java/org/apache/eventmesh/trace/zipkin/ZipkinTraceService.java:146]: OK. I will revoke this part of the modification.

**Review by xwm1992 — APPROVED** [review]

**Pull request merged**